tool/tctl/common/auth_command.go

```diff
@@ -644,6 +644,20 @@ func (a *AuthCommand) generateUserKeys(clusterAPI auth.ClientI) error {
 		return trace.Wrap(err)
 	}
 	fmt.Printf("\nThe credentials have been written to %s\n", strings.Join(filesWritten, ", "))
+
+	expires, err := key.TeleportTLSCertValidBefore()
+	if err != nil {
+		logrus.Warnf("Failed to check TTL validity: %v.", trace.Wrap(err))
+		return nil
+	}
+	certRequestTime := time.Now()
+	if certRequestTime.Add(a.genTTL).Sub(expires) > time.Minute {
```

Comment:
Couldn't we compare `expires` against the `Expires` of the `proto.UserCertsRequest` from above instead?